### PR TITLE
[Torch] Add fold rule for AtenMaskedFillTensorOp to AtenMaskedFillScalarOp

### PR DESCRIPTION
**Pull request by zhekunz2**

### CODE REVIEW

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 1]
```diff
@@ -162,6 +162,41 @@ static Value getScalarIntValue(Value input, Location loc,
...
+    return nullptr;
+
+  std::optional<unsigned> inputRank = getTensorRank(input);
+  if (!inputRank || *inputRank != 0)
+    return nullptr;
+
+  if (auto valueTensorLiteralOp = input.getDefiningOp<ValueTensorLiteralOp>()) {
+    auto val = valueTensorLiteralOp.getValue()
+                   .cast<DenseElementsAttr>()
+                   .getSplatValue<float>();
```
Comment (ramiro050) [inline lib/Dialect/Torch/IR/TorchOps.cpp:188]: I think this should be `double`, since the input type can be an `f64`

Comment (qingyunqu) [reply lib/Dialect/Torch/IR/TorchOps.cpp:188]: I think `getSplatValue<APFloat>().getValueAsDouble()` is more general for different float type.

**Review by qingyunqu — APPROVED** [review]
LGTM

**Comment by qingyunqu** [comment]
@ramiro050 Could you please review this PR again?

**Review by ramiro050 — APPROVED** [review]
Thanks!

**Pull request merged**